### ISSUE TITLE
`che-theia init` should work with commit hashes

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-che/che#17940]
### Describe the bug
When using `che-theia` I can check out out a particular commit hash, only branches. When we can't compile against current Theia master, we usually fix the build against that last commit (by hash) that che-theia still compiles against. This means that when we're in that state, our self-hosting process does not work with what we're currently shipping.

 ### Che version

 - [ ] latest
 - [x] nightly
 - [ ] other: please specify

**Comment by azatsarynnyy** [comment]
I see there are several options available:
- before `che-theia init`, checkout Theia to a commit, that Che Theia is currently aligned with, see https://github.com/eclipse/che-theia/blob/dae92d19d167c0e00c31e44630e738dc75153dd5/build.include#L16
- build the images with something like
```
./build.sh --build-args:${GITHUB_TOKEN},THEIA_COMMIT_SHA=1e09b816511769eb206494c1bac1a595e95795f9 --tag:next --git-ref:refs\\/heads\\/master --dockerfile:Dockerfile.alpine --skip-tests --branch:master
```
- just wait for QE team finish verifying https://github.com/eclipse/che/pull/17933 and  Che Theia is aligned with Theia:next https://github.com/eclipse/che-theia/pull/823

**Comment by tsmaeder** [comment]
Or we just remove the `-b` when checking out stuff. I dont' think it would make a difference. Do you see one?

**Comment by tsmaeder** [comment]
@azatsarynnyy you're right, I'm being stupid here. Still: the fact that we're forced to using branches still is an unnecessary limitation that should be removed.

**Comment by azatsarynnyy** [comment]
> Or we just remove the `-b` when checking out stuff. I dont' think it would make a difference. Do you see one?

You're right @tsmaeder. That would be a really useful improvement.

**Comment by ibuziuk** [comment]
Closing as outdated https://github.com/eclipse-che/che/issues/21771
